sdks/python/apache_beam/runners/portability/artifact_service.py

```diff
@@ -0,0 +1,122 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""Implementation of the Arrtifact{Staging,Retrieval}Service.
+
+The staging service here can be backed by any beam filesystem.
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import hashlib
+import os
+import random
+import re
+
+from apache_beam.io import filesystems
+from apache_beam.portability.api import beam_artifact_api_pb2
+from apache_beam.portability.api import beam_artifact_api_pb2_grpc
+
+
+class BeamFilesystemArtifactService(
+    beam_artifact_api_pb2_grpc.ArtifactStagingServiceServicer,
+    beam_artifact_api_pb2_grpc.ArtifactRetrievalServiceServicer):
+
+  _DEFAULT_CHUNK_SIZE = 10 << 20  # 10mb
```

Comment:
Is this tuned in any way?

Reply:
Not particularly, though this isn't super critical (as long as it isn't tiny, nor so big one runs out of memory). The Java one is 2mb, which seems a bit small, but I'll just do that.